src/carrot/core.go

```diff
@@ -50,7 +52,7 @@ func LoadTest(base *Base, latencyCh chan []float64, timeCh chan []time.Time) {
 
 	for range time.Tick(time.Millisecond * time.Duration(base.TickDelay)) {
 		routine := &Routine{time.Now(), time.Now(), 0, ""}
-		go singleTest(globalCounter, queue, base, routine)
+		go singleTest(globalCounter, queue, base, routine, lines[localCounter])
```

Comment:
Generally go shouldn't have data being passed to goroutines like this, you use channels for this thing.